### PR TITLE
Allow provide pre-defined kubernetes secret when helm-install AutoScalingRunnerSet

### PR DESCRIPTION
**Pull request by TingluoHuang**
In case the customer has created a secret with all the required data, we will just reference it instead of creating a new one.

### CODE REVIEW

**Review by rentziass — COMMENTED** [review]
Do we require the secret to follow a certain structure? Do we have a way to tell if we're dealing with a PAT or a GitHub app? (There's a good chance I'm missing something)

**Comment by TingluoHuang** [comment]
> Do we require the secret to follow a certain structure? Do we have a way to tell if we're dealing with a PAT or a GitHub app? (There's a good chance I'm missing something)

@rentziass we do, I am not quite sure how to enforce it in the helm level, so just left a comment in the [values.yaml](https://github.com/actions/actions-runner-controller/pull/2234/files#diff-7183f737213e8105334652650f1e926a560f6941b6d09fbda892a47aa4e80b8bR18)

File: charts/auto-scaling-runner-set/values.yaml [hunk 1]
```diff
@@ -13,6 +13,10 @@ githubConfigSecret:
 
   ### GitHub PAT Configuration
   github_token: ""
+## If you have a pre-define kubernetes secret in the same namespace the auto-scaling-runner-set is going to deploy,
+## you can also reference it via `githubConfigSecret: predefinedSecret`.
+## You need to make sure your predefined secret has all required secret data set properly, ex, github_token/github_app_id/github_app_installation_id/github_app_private_key
```
Comment (Link-) [inline charts/auto-scaling-runner-set/values.yaml]: nit: maybe we add an example for how to create the secret with the appropriate structure in our docs and reference it here? OR - just add the kubectl command to create the secret with the adequate structure

**Review by Link- — DISMISSED** [review]
:shipit:

**Review by Link- — DISMISSED** [review]

**Review by johnsudol — APPROVED** [review]

**Pull request merged**